Source/WebKit/NetworkProcess/storage/CacheStorageDiskStore.cpp

```diff
@@ -276,21 +282,33 @@ std::optional<CacheStorageRecord> CacheStorageDiskStore::readRecordFromFileData(
         return std::nullopt;
 
     auto headerData = fileData.subspan(metaData->headerOffset, metaData->headerSize);
-    if (metaData->headerHash != computeSHA1(headerData, m_salt))
+    if (metaData->headerHash != computeSHA1(headerData, salt))
         return std::nullopt;
 
     auto header = decodeRecordHeader(headerData);
     if (!header)
         return std::nullopt;
 
+    CacheStorageRecordInformation info { metaData->key, header->insertionTime, 0, 0, header->responseBodySize, header->request.url(), false, { } };
+    info.updateVaryHeaders(header->request, header->response.httpHeaderField(WebCore::HTTPHeaderName::Vary));
+    return StoredRecordInformation { info, *metaData, *header };
+}
+
+std::optional<CacheStorageRecord> CacheStorageDiskStore::readRecordFromFileData(const Vector<uint8_t>& buffer, const Vector<uint8_t>& blobBuffer)
+{
+    auto storedInfo = readRecordInfoFromFileData(m_salt, buffer);
+    if (!storedInfo)
+        return std::nullopt;
+
     std::optional<WebCore::DOMCacheEngine::ResponseBody> responseBody;
-    if (metaData->isBodyInline) {
-        size_t bodyOffset = metaData->headerOffset + metaData->headerSize;
-        if (bodyOffset + metaData->bodySize != fileData.size())
+    if (storedInfo->metaData.isBodyInline) {
+        auto fileData = Span { buffer.data(), buffer.size() };
```

Comment:
fileData does not seem much useful here anymore.

Reply:
It's used for getting bodyData below.

Reply:
Sure, but we could create bodyData directly without fileData.